### PR TITLE
feat: Extended the Scan API

### PR DESCRIPTION
**Pull request by umgefahren**
I have extended the scan APIs. I followed the mentioned reference implementation.

It is not tested though and it would be very kind if someone could before merging, my debug probe hasn't arrived yet.

### CODE REVIEW

**Comment by umgefahren** [comment]
Tested it, works!

File: cyw43/src/control.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use core::cmp::{max, min};
 use core::iter::zip;
+use core::time::Duration;
```
Comment (Dirbaio) [inline cyw43/src/control.rs:3]: please use `embassy_time::Duration` instead.

(`core::time::Duration` is a bit bloated for embedded usage: it has u64 seconds + u32 nanoseconds, and requires expensive multiply/divide by 10^9 when doing math with it)

File: cyw43/src/control.rs [hunk 2]
```diff
@@ -29,6 +30,45 @@ pub struct Control<'a> {
...
+#[derive(Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ScanType {
+    Active {
+        /// Period of time to wait on each channel when active scanning.
+        dwell_time: Option<Duration>,
+    },
+    Passive {
+        /// Period of time to wait on each channel when passive scanning.
+        dwell_time: Option<Duration>,
```
Comment (Dirbaio) [inline cyw43/src/control.rs:42]: maybe `dwell_time` could be a field in `ScanOptions` instead, since the two scan types have it equally?

**Comment by Dirbaio** [comment]
I've backed out the BssInfo changes because they ended up conflicting with #2461 , i'm merging the rest.

**Pull request merged**